.circleci/config.yml

```diff
@@ -43,17 +44,23 @@ common_jobs: &common_jobs
             <<: *common_filters
             requires:
               - find-develop-snapshot
+            context:
```

Comment:
> Since the CLI nightly workflow consists of multiple jobs, the PR currently proposes separate Slack notifications for the build, unit tests, and integration tests. Are there any jobs that should be added/removed?

GitHub won't let me comment on the lines I want, but is it possible to call `notify-slack` within `common_jobs`, after the `build` job? So the slack notification will alert us if the `nightly` workflow fails as a whole and we don't have to add the `notify-slack` command to multiple jobs

https://github.com/dockstore/dockstore-cli/blob/9cb046aa37932121d161d266cd66825f4e04eda1/.circleci/config.yml#L38-L43

Reply:
From what I've seen, I don't think the `slack/notify` command can assess the cumulative status of multiple jobs in a workflow (see "Is it possible to receive only the final status of the Workflow?" in https://github.com/CircleCI-Public/slack-orb/wiki/FAQ).

Seeing if we can integrate Slack via webhook instead of the orb -- it looks like CircleCI has a `workflow-completed` event (https://circleci.com/docs/webhooks/#event-specifications) which could satisfy the suggested feedback.

Reply:
Ah, I see, I wasn't sure if it was possible so you can ignore that feedback :slightly_smiling_face: 

Is it possible to add `notify-slack` to `setup_and_run_integration_tests`? It's the only command called by `non-confidential-tests`, `confidential-workflow-tests`, `confidential-tool-tests`, and `integration-tests`. That way you don't have to add it to each integration test job

https://github.com/dockstore/dockstore-cli/blob/9cb046aa37932121d161d266cd66825f4e04eda1/.circleci/config.yml#L550-L557

Reply:
I think that would work, since `notify-slack` would be a step within a job 👍 It would prevent repetition in the config file, but I guess we'd still have separate notifications for each job (as opposed to one workflow notification).

Will give it a try first, so we can decide how we feel about the multiple notifications

Reply:
It does seem a bit excessive to report with separate Slack notifications for the build, unit tests, and integration tests. 
Could consider notifying after sonar cloud which should only pass if all the other builds pass.

Reply:
A possible concern with only notifying in sonar cloud (if I'm understanding CircleCI correctly): if the build fails, the unit/integration tests (which require the build) won't run at all, and then sonar cloud (which requires those tests) won't run also? So we won't be notified at all.

To be fair, if we're only notifying on failures, the notifications shouldn't be a daily occurrence anyways

Reply:
Oh, I see. So you would be able to notify on success, but not on failure.  :thought_balloon: